config/routes.rb

```diff
@@ -11,6 +11,8 @@
   
   resources :tags
   
+  resources :comments, only: [:create, :destroy]
```

Comment:
This route does not reference the image, which is why you can't look up the image in the comments controller. You should nest this route under images, and then in your form and links it will be `image_comments_path`, or `image_comment_path(@image)`.

``` ruby
resources :images, only: [:index] do
  resources :comments, only: [:create, :destroy]
end
```

This all can optionally be nested within gallery, but does not need to and I recommend against it. Makes building URLs kind of annoying, and as long as you have the image, you find the gallery if needed by the association.